Nested folders improvements patch

bruceballad opened the pull request:
Hi, I tried to fix and improve this modified version of hashlips_art_engine. And I wanted to share my personal "improvements". However, I have never created a fork or opened a pull request before. So, please excuse the weird commits. 

Notable Changes; 

- [Fixed] A small bug with resetNameIndex functionality that breaks the numbers when it is used after second layer config set.
- [Added] The ability to add a suffix after the number in the edition name. Now, if there is a suffix, 'resetNameIndex=true' option adds the sub-index counter number after the suffix. So, you can form names like; "`PREFIX #10 - SUFFIX #2`". (If there is no suffix set, resetNameIndex works as originally intended)
- [Added] The ability to set custom descriptions for each layer config set.
- [Added] The ability to prepend custom descriptions for each layer config set. So, you can form unique descriptions like; "`Item #10 is an art piece from Collection X`".
- [Added] External_url config for OpenSea.
- [Added] Some sample extraAttributes to be used as templates based on OpenSea metadata-standards.

bruceballad committed:
resetNameIndex fix

[Fix] There is a small bug with resetNameIndex functionality that breaks the numbers when it is used after the second layer config set. Changing the "+=" to "=" at the line 556 seems to fix it.

bruceballad committed:
Improvements

- [Added] The ability to add a suffix after the number in the edition name. Now, if there is a suffix, 'resetNameIndex=true' option adds the sub-index counter number after the suffix. So, you can form names like; "PREFIX #10 - SUFFIX #2". (If there is no suffix set, resetNameIndex works as originally intended)
- [Added] The ability to set custom descriptions for each layer config set.
- [Added] The ability to prepend custom descriptions for each layer config set. So, you can form unique descriptions like; "Item #10 is an art piece from Collection X".
- [Added] External_url config for OpenSea.
- [Added] Some sample extraAttributes to be used as templates based on OpenSea metadata-standards.

bruceballad committed:
Improvements part-2

- [Added] The ability to add a suffix after the number in the edition name. Now, if there is a suffix, 'resetNameIndex=true' option adds the sub-index counter number after the suffix. So, you can form names like; "PREFIX #10 - SUFFIX #2". (If there is no suffix set, resetNameIndex works as originally intended)
- [Added] The ability to set custom descriptions for each layer config set.
- [Added] The ability to prepend custom descriptions for each layer config set. So, you can form unique descriptions like; "Item #10 is an art piece from Collection X".
- [Added] External_url config for OpenSea.
- [Added] Some sample extraAttributes to be used as templates based on OpenSea metadata-standards.

bruceballad committed:
New removeMetadataContent utility

A new utility script to remove a top level data from metadata files. It is created by modifying removeTrait.js utility.

This utility can be used for removing unwanted top level data from json files.
You can remove a data you may not need in your project,  such as "dna" or "external_url".

Use by running the following command;
```
node utils/removeMetadataContent.js "Background"
```

or for additional logging, use with the `-d` flag
```
node utils/removeMetadataContent.js "Background" -d
```

bruceballad committed:
Merge branch 'nftchef:nested-folder-structure' into nested-folders-improvements-patch

nftchef reviewed (changes requested):
Thank you so much for the PR! I left a few comments for changes before I can merge this in 🙇‍♂️

bruceballad committed:
Merge branch 'nftchef:nested-folder-structure' into nested-folders-improvements-patch

bruceballad reviewed (commented):
I agree with all your comments.
I will make the changes as soon as I have some time.

bruceballad committed:
Delete 3shadow?blend=multiply&opacity=50.png

This file name was preventing GitHub Desktop app from cloning the repo.
I know this name was experimental.
I could not try renaming it on the website too. So, I need to remove it for now.

bruceballad committed:
Changes based on the requests from nftchef

#config.js
-	The sample text of `const description =` property is returned to how it was previously.
-	`prependNameInDescription` property is removed as we now replace `{name}` key in the text for embeding names in the text anywhere we want, not only prepend.
-	The order of new properties added to the sample layer config groups are ordered for a better merge and easy reading.
-	Additional `extraAttributes` examples are removed.

#main.js
-	String combining in asset name and description texts are now all by the proper use of template strings.
-	The functions that asset names and descriptions are built are moved to a better place to reduce repeating codes.
-	Instead of passing `_prefix`, `_offset`, and `_suffix`, separately and using these during metadata writing, they are now being passed to metadata builder as just `_name`.
-	The check and switch for `_descriptionOverwrite` is now done before the description is passed to metadata builder as `_description`.
-	`_prependNameInDescription` propery is removed. Now names get embeded in the text with RegExp replacing the `{name}` key before passing the `_description` text to metadata.

#README.md
-	Explanations and examples for the new `suffix` and `descriptionOverwrite` properties are added. In the `# Name + Number prefix, suffix, and counter reset for configuration sets` section (the title is modified too).
-	A new section for `Remove Metadata Content` utility is added for directives on how to use this utility.

#3shadow_blend=multiply&opacity=50.png
-	This file name was preventing GitHub Desktop app from cloning the repo to a local disk. It was only removed for the cloning step. Now it is being added back to match the source.

bruceballad committed:
Merge branch 'nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
small comment clean up.

Deleted a leftover comment line.

bruceballad committed:
Merge branch 'nftchef:nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
Merge branch 'nftchef:nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
Merge branch 'nftchef:nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
Fix for using createPreviewCollage.js with JPEG outputs

Added a small check for the outputJPEG setting.
Now it can pull images with .jpg extension to create preview collage too.

bruceballad committed:
fix _offset console message + edition reports with name

Removed a leftover "_offset" console message while generating images.
Added "name" info in the "Created edition: #" messages in console for better reports. This will allow following prefix, suffix. and offset number changes while generating images.

bruceballad committed:
Fix for using updateBaseUri.js with JPEG outputs

The baseUri util was always making the file extension in URLs ".png".
Added a small check for the outputJPEG setting.
Now it can update baseUri and write ".jpg" or ".png" based on the config outputJPEG setting.

bruceballad committed:
adds bypass DNA

manual update reflecting the last commit made by @nftchef

https://github.com/nftchef/hashlips_art_engine/commit/2e9d74e61ae2f60f1ce33558b59dcfc28056dd22

bruceballad committed:
Merge branch 'nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
Merge branch 'nested-folder-structure' into nested-folders-improvements-patch

bruceballad committed:
Small Update

Just changed counter reset to 'true' for the second sample layerconfig set.

matbrgz commented:
Is it possible to merge this?
